honeybee_3dm/face.py

```diff
@@ -36,63 +39,115 @@ def import_faces(rhino3dm_file, tolerance=None):
     hb_shades = []
     hb_apertures = []
     hb_doors = []
-
-    # TODO: Add an input to customize layer names
-    # A Layer dictionary with layer name : (Honeybee face_type, Class) structure
-    layer_to_hb_object = {
-        'roof': (face_types.roof_ceiling, Face),
-        'wall': (face_types.wall, Face),
-        'floor': (face_types.floor, Face),
-        'airwall': (face_types.air_boundary, Face),
-        'shade': (None, Shade),
-        'aperture': (None, Aperture),
-        'door': (None, Door)
-    }
-
-    for layer in rhino3dm_file.Layers:
-        if layer.Name not in layer_to_hb_object:
-            warnings.warn(
-                f'Only objects on layers {tuple(layer_to_hb_object.keys())} will be'
-                'imported during the process of importing faces.'
-            )
-
-    # get all the objects for valid layers
-    layers = list(layer_to_hb_object.keys())
-    objects_by_layer = filter_objects_by_layers(rhino3dm_file, layers)
-
-    for layer, rhino_objects in zip(layers, objects_by_layer):
-
-        hb_face_type, hb_face_module = layer_to_hb_object[layer]
-
-        # for each rhino geometry gathered, Converting the Rhino3dm geometry
-        # into a Ladybug Face3D objects
-        for obj in rhino_objects:
-
-            lb_faces = to_face3d(obj, tolerance=tolerance)
-            name = obj.Attributes.Name
-            for face_obj in lb_faces:
-                # TODO: Double check with Chris if this naming works for energy models.
-                # if name is assigned by user use the same name for all the sub faces
-                # otherwise generate a randome name based on the layer name.
-                obj_name = name or clean_and_id_string(layer)
-                args = [clean_string(obj_name), face_obj]
-                if hb_face_type:
-                    args.append(hb_face_type)
-                    hb_face = hb_face_module(*args)
+    config = config
+    # If config file is provided and any Rhino layer is assigned
+    if config and config['HB_layers'].values():
+        # Map rhino layers to HB_layers based on the config file
+        hb_layer_from_layer = {config['HB_layers'][key]: key for key in
+            config['HB_layers']}
+        # A Layer dictionary with HB_layer : (Honeybee face_type, Class) structure
+        layer_to_hb_object = {
+            HB_layers.roof.value: (face_types.roof_ceiling, Face),
+            HB_layers.wall.value: (face_types.wall, Face),
+            HB_layers.floor.value: (face_types.floor, Face),
+            HB_layers.airwall.value: (face_types.air_boundary, Face),
+            HB_layers.shade.value: (None, Shade),
+            HB_layers.aperture.value: (None, Aperture),
+            HB_layers.door.value: (None, Door)
+        }
+        # A dictionary with child layer : parent layer structure
+        child_parent = child_parent_dict(rhino3dm_file)
+
+        for layer in rhino3dm_file.Layers:
```

Comment:
This is a very long for loop and I couldn't really follow the logic. I suggest to break down the code in smaller reusable functions/pieces.